docs/sdkx-xamarin/tracking.mdx

```diff
@@ -0,0 +1,187 @@
+
+Example:
+
+```csharp
+Dictionary<string, string> metadata = new Dictionary<string, string>();
+metadata.Add("usertype","paid");
+metadata.Add("level","7");
+metadata.Add("score","12345");
+
+Dictionary<string, object> config = new Dictionary<string, object>();
+config.Add("customMetadata", metadata);
+
+Helpshift.ShowConversation(this, config);
+```
+
+<Admonition type="info" title="Note">
+
+- Metadata should only be sent as String key-value pairs.
+- Once customMetadata is set, if you want to reset it, then you’ll have to call the SDK X API with an empty dictionary, i.e.
+
+```csharp
+Dictionary<string, string> metadata = new Dictionary<string, string>();
+Dictionary<string, object> config = new Dictionary<string, object>();
+config.Add("customMetadata", metadata);
+Helpshift.ShowConversation(this, config);
+```
+
+</Admonition>
+
+## Attaching Custom Issue Fields to conversations {#set-custom-issue-fields}
+
+You can attach Custom Issue Fields to every new conversation started by the user. A Custom Issue Field should have a key, a datatype, and a value. The Helpshift SDK allows the addition of Custom Issue Fields by using the `cifs` key in API config dictionary.
+
+These custom issue fields will be sent whenever an end user starts a new conversation or creates an issue via Smart FAQs.
+
+As soon as an end user opens the conversation screen, they see a greeting message, and the conversation is considered active. All the modified Custom Issue Fields (updated during an active conversation) will only be sent with the next conversation that end user starts.
+
+If you want to set Custom Issue Fields at the time of Issue creation, follow the steps:
+
+1. Initialise a top level custom issue fields' `Dictionary`
+2. For each custom issue field, define your custom issue field `Dictionary`
+3. Add the `"type"` and `"value"` for that custom issue field in custom issue field dictionary.
+4. Add the custom issue field dictionary to top level dictionary (with key as your configured key and value as custom issue field dictionary)
+5. Pass the dictionary to `config` with key `"cifs"` of the `Helpshift.ShowConversation(this, config)`
+
+```csharp
+Dictionary<string, object> joiningDate = new Dictionary<string, object>();
+joiningDate.Add("type", "date");
+joiningDate.Add("value", 1505927361535);
+
+Dictionary<string, string> stockLevel = new Dictionary<string, string>();
+stockLevel.Add("type", "number");
+stockLevel.Add("value", "1505");
+
+Dictionary<string, string> employeeName= new Dictionary<string, string>();
+employeeName.Add("type", "singleline");
+employeeName.Add("value", "Bugs helpshift");
+
+Dictionary<string, string> isPro = new Dictionary<string, string>();
+isPro.Add("type", "boolean");
+isPro.Add("value", "true");
```

Comment:
we should check if the value type code is correct here
"true", "1505" etc

Reply:
Yeah, only date needs to be passed as number. Rest all are strings.